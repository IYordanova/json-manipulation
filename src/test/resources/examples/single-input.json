{
  "full_name": "Edmund Monahan",
  "address": {
    "street": "755 Edna Spur",
    "city": "Florida"
  },
  "vehicles": [
    {
      "model": "2",
      "name": "Toyota Roadster",
      "manufacturer": "Lamborghini",
      "type": "Cargo Van"
    },
    {
      "model": "Alpine",
      "name": "BMW Model 3",
      "manufacturer": "Maserati",
      "type": "Crew Cab Pickup"
    },
    {
      "model": "Explorer",
      "name": "Jeep Volt",
      "manufacturer": "Lamborghini",
      "type": "Crew Cab Pickup"
    }
  ],
  "payment_details": [
    {
      "iban": "HR4286282329341701453",
      "credit_card_number": "8699454170567604",
      "currency": "Danish Krone",
      "credit_card_cvv": "753"
    },
    {
      "iban": "TR6532729HQDSUUT8JG7ZWXF41",
      "credit_card_number": "3605846184452422",
      "currency": "Seychelles Rupee",
      "credit_card_cvv": "951"
    }
  ],
  "email": "Edmund91@yahoo.com",
  "phone": "253-835-7495",
  "uid": "acf8b039-dc52-4e19-9acd-60c8cba5cc5e",
  "last_updated": "2023-10-18",
  "created_at": "2024-05-18"
}