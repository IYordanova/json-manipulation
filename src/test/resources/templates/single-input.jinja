{
    "fullName": "{{ full_name }}",
    "uid": "{{ uid }}",
    "contactDetails": {
        "address": {
            "street": "{{ address.street }}",
            "city": "{{ address.city }}"
        },
        "emailAddress": "{{ email }}",
        "phoneNumber": "{{ phone }}"
    },
    "paymentDetails": [
     {% for details in payment_details %}
        {
            "IBAN": "{{ details.iban }}",
            "cardDetails": {
                "number": "{{ details.credit_card_number }}",
                "CVV": "{{ details.credit_card_cvv }}"
            },
            "currency": "{{ details.currency }}"
        }{% if not loop.last %},{% endif %}
     {% endfor %}
    ],
    "vehicles": [
    {% for vehicle in vehicles %}
        {
            "manufacturer": "{{ vehicle.manufacturer }}",
            "model": "{{ vehicle.model }}",
            "name": "{{ vehicle.name }}",
            "type": "{{ vehicle.type }}"
        }{% if not loop.last %},{% endif %}
    {% endfor %}
    ],
    "lastUpdated": "{{ last_updated }}",
    "createdAt": "{{ created_at }}"
}