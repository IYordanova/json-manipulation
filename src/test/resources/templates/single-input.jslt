let paymentDetails = [for (.payment_details)
  {
    "IBAN": .iban,
    "cardDetails": {
        "number": .credit_card_number,
        "CVV": .credit_card_cvv
    },
    "currency": .currency
  }
]
let vehicles = [for (.vehicles)
  {
    "manufacturer": .manufacturer,
    "model": .model,
    "name": .name,
    "type": .type
  }
]
{
    "fullName": .full_name,
    "uid": .uid,
    "contactDetails": {
        "address": {
            "street": .address.street,
            "city": .address.city
        },
        "emailAddress": .email ,
        "phoneNumber": .phone
    },
    "paymentDetails": $paymentDetails,
    "vehicles": $vehicles,
    "lastUpdated": .last_updated,
    "createdAt": .created_at
}