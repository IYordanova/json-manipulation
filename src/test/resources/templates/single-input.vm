{
    "fullName": "${full_name}",
    "uid": "${uid}",
    "contactDetails": {
        "address": {
            "street": "${address.street}",
            "city": "${address.city}"
        },
        "emailAddress": "${email}",
        "phoneNumber": "${phone}"
    },
    "paymentDetails": [
    #foreach($details in $payment_details)
        {
            "IBAN": "${details.iban}",
            "cardDetails": {
                "number": "${details.credit_card_number}",
                "CVV": "${details.credit_card_cvv}"
            },
            "currency": "${details.currency}"
        }#if( $foreach.hasNext ), #end
    #end
    ],
    "vehicles": [
    #foreach($vehicle in $vehicles)
    {
        "manufacturer": "${vehicle.manufacturer}",
        "model": "${vehicle.model}",
        "name": "${vehicle.name}",
        "type": "${vehicle.type}"
    }#if( $foreach.hasNext ), #end
    #end
     ],
    "lastUpdated": "${last_updated}",
    "createdAt": "${created_at}"
}