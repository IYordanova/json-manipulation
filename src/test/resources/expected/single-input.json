{
  "fullName": "Edmund Monahan",
  "uid": "acf8b039-dc52-4e19-9acd-60c8cba5cc5e",
  "contactDetails": {
    "address": {
      "street": "755 Edna Spur",
      "city": "Florida"
    },
    "emailAddress": "Edmund91@yahoo.com",
    "phoneNumber": "253-835-7495"
  },
  "paymentDetails": [
    {
      "IBAN": "HR4286282329341701453",
      "cardDetails": {
        "number": "8699454170567604",
        "CVV": "753"
      },
      "currency": "Danish Krone"
    },
    {
      "IBAN": "TR6532729HQDSUUT8JG7ZWXF41",
      "cardDetails": {
        "number": "3605846184452422",
        "CVV": "951"
      },
      "currency": "Seychelles Rupee"
    }
  ],
  "vehicles": [
    {
      "manufacturer": "Lamborghini",
      "model": "2",
      "name": "Toyota Roadster",
      "type": "Cargo Van"
    },
    {
      "manufacturer": "Maserati",
      "model": "Alpine",
      "name": "BMW Model 3",
      "type": "Crew Cab Pickup"
    },
    {
      "manufacturer": "Lamborghini",
      "model": "Explorer",
      "name": "Jeep Volt",
      "type": "Crew Cab Pickup"
    }
  ],
  "lastUpdated": "2023-10-18",
  "createdAt": "2024-05-18"
}